modules/hamsandwich/hook.h

```diff
@@ -82,7 +82,9 @@ class Hook
 		ivtable[entry]=(int *)func;
 #if defined(_WIN32)
 		VirtualFree(tramp, 0, MEM_RELEASE);
-#elif defined(__linux__) || defined(__APPLE__)
+#elif defined(__linux__)
+		munmap(tramp, sysconf(_SC_PAGESIZE));
```

Comment:
I might be wrong, but shouldn't you pass the same size as `m_size` value? You will likely need to do something to get such length.